extensions/git/src/git.ts

```diff
@@ -1164,9 +1167,15 @@ export class Repository {
 
 		let mode: string;
 		let add: string = '';
+		let treeish: string = '';
+		const commits = await this.log({ maxEntries: 1 });
+
+		if (commits.length > 0) {
+			treeish = 'HEAD';
+		}
 
 		try {
-			const details = await this.getObjectDetails('HEAD', path);
+			const details = await this.getObjectDetails(treeish, path);
```

Comment:
`getObjectDetails` runs` lstree`, which runs `git ls-tree -l $ {treeish} -- ${path}`. I think it causes fatal error (not a valid object name) if `treeish` is empty. The try-catch block that surrounds `getObjectDetails` does not seem to handle errors except for `GitErrorCodes.UnknownPath`. Is it okay?

Reply:
`getObjectDetails` doesn't run `ls-tree` if `treeish` is empty. It runs `git ls-files --stage ...` instead. Which is what I intend in this PR because `ls-tree -l HEAD ...` will fail in an empty repo.

https://github.com/microsoft/vscode/blob/c58aaab8a1cc22a7139b761166a0d4f37d41e998/extensions/git/src/git.ts#L835-L858

Reply:
I was mistaken. Thanks for your kind feedback.